src/components/Attachments/AttachmentCarousel/extractAttachmentsFromReport.js

```diff
@@ -35,7 +42,17 @@ function extractAttachmentsFromReport(parentReportAction, reportActions) {
             if (name === 'img' && attribs.src) {
                 const expensifySource = attribs[CONST.ATTACHMENT_SOURCE_ATTRIBUTE];
                 const source = tryResolveUrlFromApiRoot(expensifySource || attribs.src);
-                const fileName = attribs[CONST.ATTACHMENT_ORIGINAL_FILENAME_ATTRIBUTE] || FileUtils.getFileName(`${source}`);
+                if (uniqueSources.has(source)) {
+                    return;
+                }
+
+                uniqueSources.add(source);
+                let fileName = attribs[CONST.ATTACHMENT_ORIGINAL_FILENAME_ATTRIBUTE] || FileUtils.getFileName(`${source}`);
+
+                // image.jpg helps render the attachment as image if we didn't obtain an extension from the source
+                if (!/\.\w+$/.test(fileName)) {
+                    fileName = 'image.jpg';
+                }
```

Comment:
public image urls might lack a file extension in the source URL (see example in PR Tests steps), without an extension our `AttachmentView` fails to recognize them as images and renders fallback content for attachments with no preview (only download)

_If this is too hacky we can try to address it in a follow-up as it would require it's own proposal and different scope of changes_

Reply:
yes hacky but agree out of scope

Reply:
alternate approach could be to not support rendering those images at all for now. But not too passionate about the order in which these problems get adressed.